pandas/core/dtypes/common.py

```diff
@@ -83,7 +83,9 @@ def ensure_float(arr):
     float_arr : The original array cast to the float dtype if
                 possible. Otherwise, the original array is returned.
     """
-    if issubclass(arr.dtype.type, (np.integer, np.bool_)):
+    if is_extension_array_dtype(arr.dtype):
+        arr = arr.to_numpy(dtype="float64", na_value=np.nan)
```

Comment:
This should do `Float32` -> `float32`, right? i.e. consider the itemsize of `arr.dtype`.

Reply:
Hmm, but only if it's already float. Also below for int/bool it's cast to "float" which means float64. I added an additional check for float dtype.